PEP 749: Use __format__ instead of .format as param name (#3993)
python/peps: peps/pep-0749.rst

However, using ``format`` as a parameter name could lead to collisions
if an annotation uses a class named ``format``. The parameter should be
positional-only and have a name that does not coincide with an
identifier used by user code. We propose to use the name ``__format__``,
as names with double underscores are reserved for the Python implementation.

> A previous version of this PEP proposed using the name ``.format`` with a
> leading dot. This has the advantage that it cannot collide with any
> identifier, reserved or not, but it also made it so ``inspect.signature``
> could not create a signature for annotate functions.

Reviewer: ```suggestion
A previous version of this PEP proposed using the name ``.format`` with a
leading dot. While ``.format`` has the advantage that it cannot collide with any
identifier, reserved or not, it also prevents ``inspect.signature``
from creating a signature for annotate functions.
```
